lib/charliecloud.py

```diff
@@ -701,6 +701,24 @@ def log(msg, hint, trace, color, prefix, end="\n"):
    if (color is not None):
       color_reset(log_fp)
 
+def monkey_write_streams():
+   """Monkey patch to replace problematic characters in stdout and stderr
+      streams when running Python 3.6. (see #1629)."""
+   def monkey_write_insert(f):
+      write_orig = f.write
+      def write_monkey(text):
+         text = text.replace("“", "\"").replace("”", "\"").replace("’", "'")
+         write_orig(text)
+      f.write = write_monkey
+   # Try to encode test string of problematic characters. If unsuccessful,
+   # monkey patch them out.
+   try:
+      "“”’".encode(encoding=sys.stdout.encoding)
+      "“”’".encode(encoding=sys.stderr.encoding)
```

Comment:
I think it's possible for these `encoding` attributes to be None, in which case this code needs a fallback.